embassy-stm32/src/opamp.rs

```diff
@@ -38,15 +38,15 @@ impl From<OpAmpSpeed> for crate::pac::opamp::vals::Opahsm {
 /// OpAmp external outputs, wired to a GPIO pad.
 ///
 /// This struct can also be used as an ADC input.
-pub struct OpAmpOutput<'d, T: Instance> {
-    _inner: &'d OpAmp<'d, T>,
+pub struct OpAmpOutput<'a, 'd: 'a, T: Instance> {
```

Comment:
these shouldn't be needed. OpAmp is covariant, meaning you can coerce from `OpAmp<'long>` to `OpAmp<'short>`, so we don't need to keep the original lifetime.

The bug is just the signatures of the functions, that should return `OpAmpOutput<'_>` instead of `OpAmpOutput<`d>`

Reply:
Ah hmmmm, yeah I see. Fixed it!